x-pack/plugins/canvas/public/components/es_data_view_select/es_data_view_select.component.tsx

```diff
@@ -31,12 +31,12 @@ export const ESDataViewSelect: React.FunctionComponent<ESDataViewSelectProps> =
   onFocus,
   onBlur,
 }) => {
-  const selectedDataView = dataViews.find((view) => value === view.title) as DataView;
+  const selectedDataView = dataViews.find((view) => value === view.title) as DataViewOption;
 
   const selectedOption = selectedDataView
-    ? { value: selectedDataView.title, label: selectedDataView.name }
+    ? { value: selectedDataView.title, label: selectedDataView.name || selectedDataView.title }
```

Comment:
💯  I was going to suggest using the nullish coalescing operator (`??`) instead, but using the logical operator is better for cases where `selectedDataView.name === ''`. 👍